docs/AutoEnforcer.md

```diff
@@ -0,0 +1,279 @@
+Autoenforcer
+=
+
+`name` implies the existence of the entry 'name' in the dicitonary. 
+
+
+### glossary
```

Comment:
* [ ] Please spellcheck this document.
* [ ] This document does not describe a global. It's kinda important and not obvious. I haven't read the whole document yet, but, at least from a user perspective, I guess it makes sense to make explicit its *scripting language* nature. We have things that look like global variables, we have things that look like functions (take inputs and return values) and so on.

Reply:
yeah the issue is that when i written that document there was not globals yet, iirc. 

maybe it's better to push back this document a bit at least until when we rolled out invalidation and c API?

Reply:
Works for me, but then expect me to push even more for Doxygen.